features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5/source/nRF5XPalGattClient.cpp

```diff
@@ -0,0 +1,1616 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017-2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+
+#include "nRF5XPalGattClient.h"
+
+#include "ble/pal/PalGattClient.h"
+#include "ble/pal/SimpleAttServerMessage.h"
+
+#include "nrf_ble_gatt.h"
+#include "nrf_ble.h"
+#include "nrf_ble_types.h"
+#include "nrf_ble_err.h"
+
+namespace ble {
+namespace pal {
+namespace vendor {
+namespace nordic {
+
+namespace {
+
+/**
+ * Extract an uint16_t value from a memory location.
+ */
+static uint16_t bytes_to_u16(const void* b)
```

Comment:
why is this needed? alignment problems?

Reply:
Yes, to avoid alignment issues. Load instructions works with unaligned data however it is not the case for Load Multiple instructions. Depending on the code, the compiler is free to use LDM instructions which raise a processor exception on unaligned data. Since it is compiler dependent, an update of the compiler, or a change in optimizations flags may issue code that don't work anymore.